### ISSUE TITLE
Keyboard focus indicator is missing - menus

### ISSUE THREAD

**Issue by briand44** [uiowa/uiowa801#9665]
# Motivation

Siteimprove is reporting "Keyboard focus indicator is missing". This is a WCAG 2.1 AA compliance issue related to some parent menu items. These menu items seem to be using `<nolink>`.

# Siteimprove info

**Technical doc**
https://alfa.siteimprove.com/rules/sia-r65

Siteimprove link to existing issue
- https://my2.siteimprove.com/Inspector/1223972/A11y/Page?pageId=83425661300&impmd=264ED37B65FA6186129334A508391970&conf=a+aa#/sia-r65/failed/Rkt6TDBGNDg=;MDUyMDBmYzM=
- https://my2.siteimprove.com/Inspector/1174032/A11y/Page?pageId=53272424901&impmd=264ED37B65FA6186129334A508391970&conf=a+aa#/sia-r65/failed/bU4zYmpLZUQ=;ZDM5ZjJlYjU=

Example
- https://obgyn.medicine.uiowa.edu/news (Siteimprove browser plugin not reporting)
- https://performingarts.uiowa.edu/about (Siteimprove browser plugin not reporting)

**Keyboard focus indicator is missing**
Issue
2.4.7: Focus Visible
This page element is not visibly highlighted when receiving focus from a keyboard.

The default focus indicator may have been removed in CSS.

**Note that this occurrence is based on a guided review.**

**Comment by briand44** [comment]
Going to close. Feel free to re-open if there was other post prod testing necessary.

**Comment by briand44** [comment]
Siteimprove is reporting this again on a couple sites:

- https://my2.siteimprove.com/Inspector/1009710/A11y/Page?pageId=26016388377&impmd=264ED37B65FA6186129334A508391970&conf=a+aa#/sia-r65/failed/b25XMEd6eUo=;YTYwYmEwNmE=
- https://my2.siteimprove.com/Inspector/1174032/A11y/Page?pageId=53272424250&impmd=264ED37B65FA6186129334A508391970&conf=a+aa#/sia-r65/failed/bU4zYmpLZUQ=;YTQwNDIyOWI=

Seems like when we addressed https://github.com/uiowa/uiowa/issues/9725, this became an issue again for top level main menu links using `<nolink>`.

**Comment by briand44** [comment]
## Grooming
 
**Quadrant:** 

Complicated

 **Discussion:**
 - issue is with top menu nav links when using `<nolink>`
 - could we use gold bottom border on focus and then gray when focus has moved to children
   - we use a light gray border bottom to designate active trail
 - Try to solve by adjusting the border bottom in these different circumstances (black or variation of gray?)
 - hopefully we can limp by with a change like ^^ until future design changes happen
 - want to test against toggle and mega menus
 - this issue doesn't appear in Siteimprove browser plugin so testing may be difficult
 - may need a separate class for after focus

The selector for the underline currently looks like this:
```
ul.sf-menu.sf-horizontal>li.sfHover>.nolink:after, 
ul.sf-menu.sf-horizontal>li.sfHover>a:after,
ul.sf-menu.sf-horizontal>li:focus>.nolink:focus:after, 
ul.sf-menu.sf-horizontal>li:focus>a:focus:after, 
ul.sf-menu.sf-horizontal>li:hover>.nolink:after, 
ul.sf-menu.sf-horizontal>li:hover>a:after {
}
```

**Proposed solution or definition of done:** 
- solve by adjusting border bottom when there is focus
- test against each menu variation (toggle, mega)
- confirm Siteimprove happy (may need to deploy in order to confirm or with https://github.com/uiowa/uiowa/pull/9698)

**Comment by bewoldt** [comment]
There's a focus indicator that adds `text-decoration: underline` to the navigation.
https://github.com/uiowa/uiowa/blob/main/docroot/themes/custom/uids_base/scss/components/menus/superfish/superfish.scss#L441-L445

Later on, the focus underline indicator on .nolink classes is removed.
https://github.com/uiowa/uiowa/blob/main/docroot/themes/custom/uids_base/scss/components/menus/superfish/superfish.scss#L460-L467

I wonder if putting `text-decoration: underline` back on will fix this.

**Comment by joewhitsitt** [comment]
I think the original idea was to differentiate a nolink from an actual link. I'd be fine with that if it fixes the issue temporarily.

Back in https://github.com/uiowa/uiowa/issues/9665, we wanted to keep the door open for larger changes coming from UIDS but it was mentioned that a pattern for nolink wasn't established there yet.

**Comment by bewoldt** [comment]
I started a separate PR #9905 for this issue. 

While doing research, I came across some other accessibility issues with the menu.

- For the Horizontal/Mega mobile and Toggle menu. When you tab through the menu and expand one of the level 1 menus, the submenu appears broken and is not displayed as block. The level 1 `<li>` is not getting the class of sf-expanded.
- For the Horizontal/Mega mobile and Toggle menu. When you tab into the menu the sf-depth-2 nolink and anchors do not show the :before dropdown indicator arrow. (when tabbing through site)
- For the Horizontal/Mega mobile and Toggle menu. The sf-depth-2 anchors and nolinks do not have padding-left: 30px (when tabbing through site)
- For the Mega mobile and Toggle menu. When you tab through the menu and expand one of the main level menus, the parent clone menu does not show.
- Aria-expanded is not getting updated when the menu accordion/dropdown is opened. It is set as false when opened or closed.
- There are also some style differences between the Horizontal/Mega/Toggle menus. The Horizontal mobile version has `<li>` as font-weight: 400 but Toggle/Mega are set to bold.

**Root cause**: sfsmallscreen.js only has a click event handler — no keyboard events
**Result**: Tabbing through the mobile accordion menus with a keyboard does not trigger .sf-expanded and the cloned parent link is never inserted.

<img width="558" height="413" alt="Image" src="https://github.com/user-attachments/assets/69002136-fa81-418a-a5ed-c873a291c176" />

<img width="557" height="509" alt="Image" src="https://github.com/user-attachments/assets/3696c5b0-ee83-4c2f-a77c-f27bd3e5a4f6" />

**Comment by joewhitsitt** [comment]
@bewoldt thanks for documenting all these. I think they might belong in https://github.com/uiowa/uiowa/issues/3517 but also increases the reason to consider moving off superfish (https://github.com/uiowa/uiowa/issues/4973).

We have several superfish related issues https://github.com/uiowa/uiowa/issues?q=state%3Aopen%20label%3Asuperfish

**Comment by briand44** [comment]
I rescanned a couple of the pages that were reporting this issue and they appear to be resolved now with this fix. I know we have a follow up that hasn't been deployed yet but wanted to note that Siteimprove is happy with this fix!
